languages/cpp/src/main/java/de/jplag/cpp/experimental/UnreachableCodeFilter.java

```diff
@@ -0,0 +1,138 @@
+package de.jplag.cpp.experimental;
+
+import static de.jplag.SharedTokenType.FILE_END;
+import static de.jplag.cpp.CPPTokenType.*;
+
+import java.util.List;
+import java.util.ListIterator;
+
+import de.jplag.TokenType;
+
+/**
+ * Contains a basic algorithm for detecting tokens contained in unreachable code.
+ */
+public final class UnreachableCodeFilter {
+
+    private UnreachableCodeFilter() {
+    }
+
+    /**
+     * Applies the filtering on the provided token list.
+     * @param tokenList The list that will be filtered. The contents of this parameter will be modified.
+     */
+    public static void applyTo(List<de.jplag.Token> tokenList) {
```

Comment:
Any reason for the fully qualified name instead of an import?

Reply:
There is a `de.jplag.cpp.Token` and a `de.jplag.Token` class, so I originally put it there for clarity.
But since only one of the two classes is used here, the import suffices and this is indeed redundant.
I will remove the fully qualified names.